cylc/flow/scripts/lint.py

```diff
@@ -0,0 +1,583 @@
+    count = 0
+    for line_no, line in enumerate(lines, start=1):
+        for check, message in checks.items():
+            # Tests with for presence of Jinja2 if no shebang line is
+            # present.
+            if (
+                jinja_shebang
+                and message['short'].startswith(
+                    JINJA2_FOUND_WITHOUT_SHEBANG)
+            ):
+                continue
+            if check.findall(line) and not line.strip().startswith('#'):
+                count += 1
+                if modify:
+                    if message['url'].startswith('http'):
+                        url = message['url']
+                    else:
+                        url = URL_STUB + message['url']
+                    outlines.append(
+                        f'# [{message["purpose"]}{message["index"]:03d}]: '
+                        f'{message["short"]}\n'
+                        f'# - see {url}'
+                    )
+                else:
+                    print(
+                        Fore.YELLOW +
+                        f'[{message["purpose"]}{message["index"]:03d}]'
+                        f'{file_}: {line_no}: {message["short"]}'
+                    )
+        if modify:
+            outlines.append(line)
+    if modify:
+        file_.write_text('\n'.join(outlines))
+    return count
+
+
+def get_cylc_files(base: Path) -> Generator[Path, None, None]:
+    """Given a directory yield paths to check.
+    """
+    excludes = ['log']
+
+    for rglob in FILEGLOBS:
+        for path in base.rglob(rglob):
+            # Exclude log directory:
+            if path.relative_to(base).parts[0] not in excludes:
+                yield path
+
+
+def get_reference(checks):
+    """Print a reference for checks to be carried out.
+
+    Returns:
+        RST compatible text.
+    """
+    output = ''
+    current_checkset = ''
+    for check, meta in checks.items():
+        # Check if the purpose has changed - if so create a new
+        # section title:
+        if meta['purpose'] != current_checkset:
+            current_checkset = meta['purpose']
+            title = CHECKS_DESC[meta["purpose"]]
+            output += f'\n{title}\n{"-" * len(title)}\n\n'
+
+        # Fill a template with info about the issue.
+        template = (
+            '{checkset}{index:003d} ``{title}``:\n    {summary}\n'
+            '    see - {url}\n'
+        )
+        if meta['url'].startswith('http'):
+            url = meta['url']
+        else:
+            url = URL_STUB + meta['url']
+        msg = template.format(
+            title=check.pattern.replace('\\', ''),
+            checkset=meta['purpose'],
+            summary=meta['short'],
+            url=url,
+            index=meta['index'],
+        )
+        output += msg
+    output += '\n'
+    return output
+
+
+def get_option_parser() -> COP:
+    parser = COP(
+        __doc__,
+        argdoc=[('[DIR ...]', 'Directories to lint')],
```

Comment:
The centralised way of having an optional arg to ensure consistency
```suggestion
        argdoc=[COP.optional(('DIR ...', 'Directories to lint'))],
```